### PR TITLE
Added support for remaining endpoints

### PR DESCRIPTION
**Pull request by JasonStoltz**
This PR adds support for the remaining APIs that were not previously covered.

They are:
  1. Clickthrough: https://swiftype.com/documentation/app-search/api/clickthrough
  2. Synonyms: https://swiftype.com/documentation/app-search/api/synonyms
  3. Analytics: https://swiftype.com/documentation/app-search/api/analytics
  4. Credentials: https://swiftype.com/documentation/app-search/api/credentials
  5. Schema: https://swiftype.com/documentation/app-search/api/schema
  6. Curations: https://swiftype.com/documentation/app-search/api/curations
  7. Logs: https://swiftype.com/documentation/app-search/api/logs

The Credentials API requires an Admin API Key to operate. An Admin API Key can't really be used for anything else. For this reason, running specs now requires an additional Environment Variable to be set: `AS_ADMIN_KEY`.

I have not tried to make any improvements in this PR. The code I have added is meant to maintain the status quo for the project. It is very simplistic and simply follows existing patterns.

The tests I have added are very basic integration tests. Since the API clients are mainly just passthrough, I simply ensure that the API endpoint can be hit successfully with a full set of parameters.

The one optimization I *did* make in this PR, is I refactored the single "client_spec.rb" file into individual specs.

The naming conventions used, and descriptions in comments are based on https://github.com/elastic/app-search-php/blob/master/resources/api/api-spec.yml. The only exception is that in the ruby client we use the term "destroy" rather than "delete".

### CODE REVIEW

File: lib/elastic/app-search/client/engines.rb [hunk 1]
```diff
@@ -21,6 +22,12 @@ def create_engine(engine_name, language = nil)
         def destroy_engine(engine_name)
           delete("engines/#{engine_name}")
         end
+
+        def update_documents(engine_name, documents)
```
Comment (yakhinvadim) [inline lib/elastic/app-search/client/engines.rb]: I'm not sure this method should be in engines, it looks like it's a separate API https://swiftype.com/documentation/app-search/api/documents

File: lib/elastic/app-search/client/credentials.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+module Elastic
+  module AppSearch
+    class Client
+      module Credentials
+
+        # Retrieve available credentials
+        def list_credentials(current: 1, size: 20)
+          get("credentials", :page => { :current => current, :size => size })
+        end
+
```
Comment (yakhinvadim) [inline lib/elastic/app-search/client/credentials.rb]: Looks like getOneKey method is missing https://swiftype.com/documentation/app-search/api/credentials#single

File: spec/credentials_spec.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    let(:updated_api_key) do
+      api_key['write'] = true
+      api_key
+    end
+
+    before { client.create_credential(api_key) }
+    after { client.destroy_credential(key_name) }
+    subject { client.update_credential(key_name, updated_api_key) }
+
+    it 'will create an API Key' do
```
Comment (yakhinvadim) [inline spec/credentials_spec.rb]:
```suggestion
    it 'will update an API Key' do
```

File: spec/credentials_spec.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      expect(subject['write']).to(eq(true))
+    end
+  end
+
+  context '#list_credentials' do
+    before { client.create_credential(api_key) }
+    after { client.destroy_credential(key_name) }
+    subject { client.list_credentials }
+
+    it 'will create an API Key' do
```
Comment (yakhinvadim) [inline spec/credentials_spec.rb]:
```suggestion
    it 'will list all API Keys' do
```

**Comment by JasonStoltz** [comment]
> Would you mind creating separate PRs for refactoring and new features in the future, please? 

Yes, sorry about that. Thank you for wading through it anyway 🙏. Sorry about all of the typos, also.

File: lib/elastic/app-search/client/synonyms.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+          get("engines/#{engine_name}/synonyms/#{id}")
+        end
+
+        # Create a new synonym set
+        def create_synonym_set(engine_name, synonyms)
+          post("engines/#{engine_name}/synonyms", :synonyms => synonyms)
+        end
+
+        # Update an existing synonym set
+        def update_synonym_set(engine_name, id, synonyms)
```
[the remark this answers is not in the document]
Comment (JasonStoltz) [reply lib/elastic/app-search/client/synonyms.rb]: I suppose it is. I will add a ticket to documentation and php client: https://github.com/elastic/app-search-php/blob/master/resources/api/api-spec.yml

[the remark this answers is not in the document]
Comment (JasonStoltz) [reply lib/elastic/app-search/client/synonyms.rb]: https://github.com/elastic/app-search-php/pull/18
https://github.com/elastic/swiftype-pubs/issues/390

File: README.md [hunk 6]
```diff
@@ -292,10 +289,273 @@ client = Elastic::AppSearch::Client.new(host_identifier: 'host-c5s2mj', api_key:
...
+
+client.get_count_analytics(engine_name, options)
+```
+
+#### Creating Synonym Sets
+
+```ruby
+engine_name = 'us-national-parks'
+
+client.create_synonym_set(engine_name, ['park', 'trail'])
```
[the remark this answers is not in the document]
Comment (JasonStoltz) [reply README.md]: Done.

File: spec/click_spec.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
+describe Elastic::AppSearch::Client::Click do
+  include_context 'App Search Credentials'
+  include_context 'Test Engine'
+
+  let(:client) { Elastic::AppSearch::Client.new(client_options) }
+
+  context '#log_click_through' do
+    let(:documents_response) { client.index_documents(engine_name, documents) }
+    let(:documents) { [first_document, second_document] }
+    let(:request_id) { 'id' }
```
[the remark this answers is not in the document]
Comment (JasonStoltz) [reply spec/click_spec.rb]: Removed.

File: spec/click_spec.rb [hunk 8]
```diff
@@ -0,0 +1,30 @@
+describe Elastic::AppSearch::Client::Click do
+  include_context 'App Search Credentials'
+  include_context 'Test Engine'
+
+  let(:client) { Elastic::AppSearch::Client.new(client_options) }
+
+  context '#log_click_through' do
+    let(:documents_response) { client.index_documents(engine_name, documents) }
```
[the remark this answers is not in the document]
Comment (JasonStoltz) [reply spec/click_spec.rb]: Updated.

File: lib/elastic/app-search/client/engines.rb (hunk 1 above)
Comment (JasonStoltz) [reply lib/elastic/app-search/client/engines.rb]: Looks like a freeloader 😠

File: lib/elastic/app-search/client/credentials.rb (hunk 2 above)
Comment (JasonStoltz) [reply lib/elastic/app-search/client/credentials.rb]: Added in https://github.com/elastic/app-search-ruby/pull/16/commits/9c5273ede4a05e428d6b89c1914d9744a8e18c4e

**Comment by JasonStoltz** [comment]
@yakhinvadim Everything should be addressed and ready for a second look. TY.

File: README.md [hunk 9]
```diff
@@ -292,10 +289,273 @@ client = Elastic::AppSearch::Client.new(host_identifier: 'host-c5s2mj', api_key:
...
+
+client.destroy_synonym_set(engine_name, 'syn-5d8e6b5d40caae7dcb6e1b9c')
+```
+
+#### Listing Credentials
+
+```ruby
+client.list_credentials(:current => 1, :size => 20)
+```
+
```
Comment (yakhinvadim) [inline README.md]: We need to document the new get_credential method

**Comment by yakhinvadim** [comment]
Other from that everything looks good

File: README.md (hunk 9 above)
Comment (JasonStoltz) [reply README.md]: https://github.com/elastic/app-search-ruby/pull/16/commits/fd8f37435e75bd8452416aa267bfc9b734975fe7

**Comment by JasonStoltz** [comment]
@yakhinvadim Should be good now.

**Pull request merged**